maintenance/wikia/migrateImagesToGcs.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use Wikia\Rabbit\ConnectionBase;
+
+require_once( __DIR__ . '/../Maintenance.php' );
+
+class MigrateImages extends Maintenance {
+
+	const TABLE_IMAGE = 'image';
+	/** @var DatabaseBase */
+	private $db;
+	/** @var LocalRepo */
+	private $repo;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Migrate images to GCS";
+	}
+
+	public function execute() {
+		$this->db = wfGetDB( DB_SLAVE );
+		// This means that this script should be run for communities for which we have not switched to GCSFileBackend
+		$this->repo = RepoGroup::singleton()->getLocalRepo();
+
+		( new \WikiaSQL() )->SELECT( "page.*, revision.*" )
+			->FROM( 'page' )
+			->LEFT_JOIN( 'revision' )
+			->ON( 'revision.rev_page = page.page_id' )
+			->WHERE( 'page.page_namespace' )
+			->EQUAL_TO( NS_FILE )
+			->runLoop( $this->db, function ( &$pages, $row ) use ( &$display_order ) {
+				$this->publishImageUploadRequest( $this->getFile( $row ), $row->page_id,
+					$row->rev_id, $row->rev_user );
+			} );
+	}
+
+	private function getFile( $row ): LocalFile {
+		if ( $row->page_latest === $row->rev_id ) {
+			return $this->repo->newFile( $row->page_title );
+		} else {
+			return $this->repo->newFile( $row->page_title, $row->rev_timestamp );
+		}
+	}
+
+	private function publishImageUploadRequest( LocalFile $file, $pageId, $revisionId, $uploaderId ) {
+		global $wgGoogleCloudUploaderPublisher;
+
+		$sha1 = $this->repo->getFullSha1( $file->getPath() );
```

Comment:
I'm guessing `image.img_sha1` is not suitable here, right?

Reply:
Yeah, img_sha1 is a weird hash of a sha1 and we've decided to store full sha1 which we can always hash.

Reply:
kk